sanic/response.py

```diff
@@ -112,8 +113,16 @@ async def send(
         """
         if data is None and end_stream is None:
             end_stream = True
-        if end_stream and not data and self.stream.send is None:
-            return
+        if self.stream.send is None:
+            if end_stream and not data:
+                return
+            elif self.stream.stage == Stage.IDLE:
+                raise ResponseException(
+                    "Response stream was ended, no more response data is "
+                    "allowed to be sent."
+                )
+            else:
+                raise SanicException("Send response function isn't available")
```

Comment:
Minor style issues. 'elif' and 'else' are unnecessary because each case returns/raises. I suggest two regular ifs and nothing on the final part (when does the final one occur, anyway? -- if it doesn't, just give the stream has ended exception without any test). Also `is` comparison for Stage enum.

Reply:
Thanks, fixed.

Reply:
Actually, if the last raising removed, type checks will fail because None is not callable. So, I would propose make a message string (e4b53a01c2f633fbfea1a1d5b580fc457bf0cc1b), wondering whether it looks okay.